scripts/gen_syscall_header.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+
+def gen_macro(ret, argc, inline):
+    sys.stdout.write("K_SYSCALL_DECLARE%d%s%s(id, name" % (argc,
+        ("" if ret else "_VOID"), ("_INLINE" if inline else "")))
+    if (ret):
+        sys.stdout.write(", ret")
+    for i in range(argc):
+        sys.stdout.write(", t%d, p%d" % (i, i))
+    sys.stdout.write(")")
+
+def gen_fn(ret, argc, name, extern):
+    sys.stdout.write("\t%s %s %s(" %
+            (("extern" if extern else "static inline"),
+             ("ret" if ret else "void"), name))
+    if argc == 0:
+        sys.stdout.write("void");
+    else:
+        for i in range(argc):
+            sys.stdout.write("t%d p%d" % (i, i))
+            if i != (argc - 1):
+                sys.stdout.write(", ")
+    sys.stdout.write(")")
+
+def gen_make_syscall(ret, argc):
+    if (ret):
+        sys.stdout.write("return (ret)")
+    if (argc <= 6):
+        sys.stdout.write("_arch")
+    sys.stdout.write("_syscall_invoke%d(" % (argc))
+    for i in range(argc):
+        sys.stdout.write("(u32_t)p%d, " % (i))
+    sys.stdout.write("id); \\\n")
+
+def gen_call_impl(ret, argc):
+    if (ret):
+        sys.stdout.write("return ")
+    sys.stdout.write("_impl_##name(")
+    for i in range(argc):
```

Comment:
No need to change this loop, but you can write this as: `sys.stdout.write(", ".join("p%d" % i for i in range(argc)))`.